styleguide/drafts/tooling.rst

```diff
@@ -0,0 +1,518 @@
+      - Fairly new, some edge cases my not supported yet
+    * - Uses pyproject.toml for configuration
+      - Toml file definition is less flexible than setup.py based one
+    * - Takes care of project versioning
+      - Editable and/or repo based install(s) are not possible "out of the box"?
+    * - Dependency pinning
+      - setup.py is still required for installations with older versions of pip
+    * - Very active development & community
+      -
+    * - Good & Powerful CLI
+      -
+    * - No manual dependency editing required
+      -
+
+Alternatives
+~~~~~~~~~~~~
+
+* `pipenv <https://pipenv.pypa.io/en/latest/>`_
+
+
+Code Health
+___________
+
+.. figure:: https://imgs.xkcd.com/comics/code_quality.png
+   :alt: xkcd-2347
+   :target: https://xkcd.com/2347/
+
+   source: `xkcd.com <xkcd_>`_
+   license: `CC BY-NC 2.5`_
+
+Pylint_
++++++++
+`Project Homepage <Pylint_www_>`_
+
+Helps finding bugs and issues before they are a problem and improves overall code quality.
+
+Pros & Cons
+~~~~~~~~~~~
+
+.. list-table::
+    :header-rows: 1
+    :widths: 50 50
+
+    * - Pros ✅
+      - Cons ❌
+    * - Lots of good warnings and hints
+      - Extra dependency
+    * - Extension mechanism (custom extensions e.g. perflint)
+      - Extra learning curve not all messages maybe straight forward
+    * - Rating check 0-10 makes it possible to steadily improve a code base
+      -
+    * - Compatible with sonar
+      -
+    * - Supports pyproject.toml
+      -
+
+Alternatives
+~~~~~~~~~~~~
+
+* `Flake8 <https://flake8.pycqa.org/en/latest/>`_
+* `pydocstyle <http://www.pydocstyle.org/en/stable/>`_
+* `Radon <https://radon.readthedocs.io/en/latest/>`_
+
+Coverage_
++++++++++
+`Project Homepage <Coverage_www_>`_
+
+Pros & Cons
+~~~~~~~~~~~
+
+.. list-table::
+    :header-rows: 1
+    :widths: 50 50
+
+    * - Pros ✅
+      - Cons ❌
+    * - Threshold can be asserted if "wanted"
+      - Extra dependency
+    * - Various output formats, compatible with coveralls.io
+      -
+    * - Supports pyproject.toml
+      -
+    * - Plugin mechanism available
+      -
+
+Alternatives
+~~~~~~~~~~~~
+
+* ¯\\_(ツ)_/¯
+
+MyPy_
++++++
+`Project Homepage <MyPy_www_>`_
+
+Having type hints (information) is nice help, but having it actually checked finds errors and makes sure the type hints are correct.
+
+Pros & Cons
+~~~~~~~~~~~
+
+.. list-table::
+    :header-rows: 1
+    :widths: 50 50
+
+    * - Pros ✅
+      - Cons ❌
+    * - Enforced type checking makes sure type annotations are correct
+      - Extra configuration, makes workspace more complex
+    * - Prevents potential bugs
+      - Projects without any type annotations so far need migration strategy
+
+Alternatives
+~~~~~~~~~~~~
+
+* `pytype <https://github.com/google/pytype>`_ (google)
+* `pyright <https://github.com/Microsoft/pyright>`_ (microsoft)
+* `pyre-check <https://github.com/facebook/pyre-check>` (facebook, contains security checking too)
+* IDE built in e.g. PyCharm
+
+PreCommit_
```

Comment:
I thought this is part of automation

Reply:
Mostly yes, but it would be a "dev-toosl" dependency that's why I have it included here.

Reply:
But wouldn't be nox also a dev-tool? So, I would mention it after nox.

Reply:
Ok understood, makes sense I'll do.